docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,741 @@
+# Rework of Mesh*Routes matching
+
+* Status: accepted
+
+Technical story: https://github.com/kumahq/kuma/issues/10247
+
+## Context and Problem Statement
+
+At the moment when targeting `Mesh*Route` you need to put it in topLevel targetRef. 
+We want to iterate over policy design and move `Mesh*Route` to `spec.to[].targetRef` section follow model where `spec.targetRef`
+selects whole proxies, `spec.from[].targetRef` selects inbounds and `spec.to[].targetRef` selects outbounds
+
+## Decision Outcome
+
+- add possibility to put `Mesh*Routes` in `spec.to[].targetRef`
+
+## Considered options
+
+1. add possibility to put `Mesh*Routes` in `spec.to[].targetRef` -> `spec.targerRef` selects proxies, `spec.from[].targetRef` selects inbounds, `spec.to[].targetRef` selects outbounds.
+2. leave `Mesh*Routes` in top level targetRef as it is now, and introduce `spec.workloadTargetRef` to match whole proxies
+
+### Add possibility to put Mesh*Routes in `spec.to[].targetRef`
+
+#### Advantages
+
+- simplifying policy api as `spec.targetRef` will always pick proxy and `spec.to[].targetRef` section will select outgoing traffic
+- ability to set different confs on different proxy for the same route (overriding default conf for route without producer/consumer policies, on universal)
+- no need for specifying `spec.targetRef` when targeting `Mesh`, simplifies policy
+- gives more flexibility as you can target multiple outbounds in single policy. Reducing number of consumer policies
+- more consistent with current approach
+- easier policy matching code
+
+#### Disadvantages
+
+- can be harder to understand because `spec.targetRef` can be MeshSubset (which is optional and probably will rarely be used)
+- yet another migration for users (only for system policies as for namespaced policies this is already not allowed) 
+- mixing outbounds and routes in `spec.to[].targetRef`
+- hides the most often most important information under `to[]` (only in certain cases as most of the time policy will target Mesh and will have only `spec.to[].targetRef`)  
+
+### Leave Mesh*Routes in top level targetRef as it is now and introduce `spec.workloadTargetRef`
+
+#### Advantages
+
+- can be simpler with clear distinction of selecting proxies in `spec.workloadTargetRef` and other resources in `spec.targetRef`
+
+#### Disadvantages
+
+- mixing proxies, outbounds and routes in `spec.targetRef`
+- when `Mesh*Route` is in `spec.targetRef` targetRef we still have `spec.to[].targetRef` section which can be misleading as route already points to some outbound
+- cannot configure multiple outbounds in the same policy, forcing users to create and later manage multiple policies, which can be problematic with consumer policies
+- MeshService in topLevel can be misleading. Is it targeting all proxies belonging to MeshService or outbound to MeshService?
```

Comment:
Hmm I don't think it's misleading anymore. `MeshService` (the new resource) represents a destination. These semantics are why `HTTPRoute` uses `Service` as `spec.parentRef`.

Reply:
What we should also be asking is "how does `spec.from` work in this option"? The answer being that we get rid of `from` as well. And we have `MeshTrafficPermission` for example target `spec.targetRef.kind: WorkloadIdentity`.

Reply:
I've updated this point to make it clear that this could be problematic for existing users.

Reply:
As for the second point I am not sure if we want to cover this here, like at the begging I thought that decision was already made, and I am just writing MADR on how to introduce this change, now this MADR is changing into fundamental redesign of policy API. I don't know what to focus on any more